doc/compressed_transactions.md

```diff
@@ -0,0 +1,152 @@
+
+The main applications for this schema are for steganography, satellite/radio broadcast, and
+other low bandwidth channels with a high CPU availability on decompression. We
+assume users have some ability to shape their transactions to improve their
+compressibility, and therefore give special treatment to certain transaction forms.
+
+## 2. Methods
+
+The four main methods to achieve a lower transactions size are:
+
+1. packing transaction metadata before the transaction and each of its inputs and
+outputs to determine the structure of the following data.
+2. replacing 32-bit numeric values with variable-length integers (VarInts).
+3. using compressed signatures and public key recovery upon decompression.
+4. replacing the 36-byte txid/vout pair with a blockheight and output index.
+
+Method 4 will cause the compressed transaction to be undecompressable if a block
+reorg occurs at or before the block it's included in. Therefore, we'll only compress
+the Txid if the transaction input is at least one hundred blocks old.
+
+## 3 Schema
+
+### 3.1 Primitives
+
+| Name             | Width     | Description |
+|------------------|-----------|-------------|
+| VarInt           | 1-5 Bytes | For 0-253, encode the value directly in one byte. For 254-65535, encode 254 followed by 2 little-endian bytes. For 65536-(2^32-1), encode 255 followed by 4 little-endian bytes. This is the same varint used to length-prefix strings in the ordinary transaction encoding. |
+| VarInt Flag      | 2 Bits    | 1, 2 or 3 indicate literal values. 0 indicates that the value will be encoded in a later VarInt. |
+| Bitstream VarInt | 1-8       | 7-bit little-endian encoding, with each 7-bit word encoded in a byte. The highest bit of each byte is 1 if more bytes follow, and 0 for the last byte. |
+| VLP-Bytestream   | 2-- Bytes | A VarInt Length Prefixed Bytestream, Has a VarInt prefixed to determine the length. |
+
+### 3.2 General Schema
+
+| Name                  | Width              | Description |
+| ----------------------|--------------------|-------------|
+| Transaction Metadata  | See Section 3.3    | Information on the structure of the transaction. See section 3.3. |
+| Minimum Blockheight   | 1-8 Bytes          | A Bitstream VarInt containing the Minimum Blockheight of which the transaction locktime and input blockheights are given as offsets. |
+| Version               | 0-5 Bytes          | An optional VarInt containing the transactions version. |
+| Input Count           | 0-5 Bytes          | An optional VarInt containing the transactions input count. |
+| Output Count          | 0-5 Bytes          | An optional VarInt containing the transactions output count. |
+| LockTime              | 0-5 Bytes          | An optional VarInt containing the transaction LockTime if its non zero. |
+| Input/Output Metadata | See Sections 3.4-5 | A Bitstream VarInt containing metadata on the inputs and outputs of the transaction. See section 3.4 and 3.5. |
```

Comment:
In ec497e8c8cd3ec844f1a482f3e64748d544c8a3a:

We should combine sections 3.4 and 5 (see my below comment) and we need to clarify that this is a "bitstream varint" where each 7-bit word indicates an input/output pair rather than part of a larger number. And that the 1-9 size limit does not apply. And explain what happens if the number of inputs and outputs differs.

Reply:
Or is it that we encode all the input metadata followed by all the output metadata? I forget where we landed here. In any case it needs to be in the spec :)

Reply:
Fixed, Thanks!